### PR TITLE
(INTENG-4572) V2 Event updates

### PR DESCRIPTION
**Pull request by christy87**
1. Added support for Segment#reset()
Segment#Reset()  - > Branch#logout()

2. Added support for Segment#alias()
Segment#alias ->  Branch#setIdentity()

3. Missing Segment - Branch Key mapping
- BUO#CanonicalIdentifier

- BUO#CanonicalUrl

-BUO#ContentImgUrl

4. Fixing SegmentEvent#product[] related issue

Segment event may have fields which are common for events and the
product associated with the events(price, sku etc). If there is product
specific value use that for BUO

5. Fixing SegmentEvent#product[] related issue

TestBed update for new functionalities

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,58 +1,24 @@
 buildscript {
   repositories {
-    mavenCentral()
+      jcenter()
+      google()
   }
-
   dependencies {
-    classpath 'com.android.tools.build:gradle:1.3.1'
+    classpath 'com.android.tools.build:gradle:3.2.1'
```
Comment (csalmi-branch) [inline build.gradle]: Why did you choose this version instead of the most recent stable version (3.3.1)?

File: gradle.properties [hunk 2]
```diff
@@ -17,3 +17,5 @@ POM_SCM_DEV_CONNECTION=scm:git:ssh://git@github.com:BranchMetrics/Segment-Branch
 POM_LICENCE_NAME=The MIT License (MIT)
 POM_LICENCE_URL=http://opensource.org/licenses/MIT
 POM_LICENCE_DIST=repo
+POM_DEVELOPER_ID=branch
```
Comment (csalmi-branch) [inline gradle.properties]: Won't we need to bump the version number in the gradle properties as well? I mean, we can do that as a separate release, but it might be better to have it here if we are just going to release all of these changes as soon as this is merged.

File: src/main/java/com/segment/analytics/android/integrations/branch/BranchIntegration.java [hunk 3]
```diff
@@ -1,40 +1,66 @@
...
 import io.branch.referral.Branch;
+import io.branch.referral.Defines;
+import io.branch.referral.util.BRANCH_STANDARD_EVENT;
+import io.branch.referral.util.BranchEvent;
+import io.branch.referral.util.CurrencyType;
 
 /**
  * @author sahilverma
- * @date 2015-11-05
+ * 2015-11-05
```
Comment (csalmi-branch) [inline src/main/java/com/segment/analytics/android/integrations/branch/BranchIntegration.java]: Why did you remove the @date tag?

File: gradle.properties (hunk 2 above)
Comment (Wiley-1) [reply gradle.properties]: The dev ID and dev Name  were missing in the staging and it causes compilation issue.
Yes. Version number needed to be bumped up before release  but do we need it to add it feature PR or to master at time of release as mentioned in Readme? ( My suggestion is to add on master at the time of release assuming there will be more features)

File: src/main/java/com/segment/analytics/android/integrations/branch/BranchIntegration.java (hunk 3 above)
Comment (Wiley-1) [reply src/main/java/com/segment/analytics/android/integrations/branch/BranchIntegration.java]: This was causing a java doc warning since @date is not a standard java doc tag

File: build.gradle (hunk 1 above)
Comment (Wiley-1) [reply build.gradle]: This need basic gradle toto updated to 4.10.1.from current version is 4.6 . The deep link SDK is using 4.6 so thought of following the same
